addons/stock_delivery/models/delivery_carrier.py

```diff
@@ -107,6 +107,8 @@ def _get_packages_from_order(self, order, default_package_type):
             total_cost += self._product_price_to_company_currency(line.product_qty, line.product_id, order.company_id)
 
         total_weight = order._get_estimated_weight() + default_package_type.base_weight
+        order_weight = self.env.context.get('order_weight', False)
+        total_weight = order_weight if order_weight and total_weight != order_weight else total_weight
```

Comment:
```suggestion
        total_weight = self.env.context.get('order_weight', False)

```
You are sending order_weight in both condition then why don't you always pass order_weight